File: cpp/nop_benchmark.cpp
// nop_benchmark.cpp
// C++ program to count NOPs (empty inline asm) executed in 1 second
#include <iostream>
#include <ctime>

int main() {
    volatile unsigned long long nops = 0;
    std::time_t start = std::time(nullptr);
    while (std::time(nullptr) - start < 1) {
        __asm__ __volatile__("nop");
        nops++;
    }
    std::cout << "NOPs executed: " << nops << std::endl;
    return 0;
}

File: python/nop_benchmark.py
# nop_benchmark.py
# Python program to count 'NOPs' (empty pass) executed in 1 second
import time

nops = 0
start = time.time()
while time.time() - start < 1:
    pass
    nops += 1
print(f"NOPs executed: {nops}")
